Firebase.h

```diff
@@ -25,49 +25,97 @@
 #include <WiFiClientSecure.h>
 #include <ESP8266HTTPClient.h>
 
-// FirebaseError represents a Firebase API error with a code and a
-// message.
-class FirebaseError {
- public:
-  operator bool() const { return _code < 0; }
-  int code() const { return _code; }
-  const String& message() const { return _message; }
-  void reset() { set(0, ""); }
-  void set(int code, const String& message) {
-    _code = code;
-    _message = message;
-  }
- private:
-  int _code = 0;
-  String _message = "";
-};
+//TODO(edcoyne) split these into multiple files.
+
+class FirebaseCall;
+class FirebaseEventStream;
 
-// Firebase is the connection to firebase.
+// Primary client to the Firebase backend.
 class Firebase {
  public:
   Firebase(const String& host);
   Firebase& auth(const String& auth);
-  const FirebaseError& error() const {
-    return _error;
+
+  // Fetch result at "path" to a local variable. If the value is too large you will exceed
+  // local memory.
+  FirebaseCall get(const String& path);
```

Comment:
another thing is that if we return a copy of the FirebaseCall object, user do have to worry about its lifetime:
i.e: if you do Firebase.get().rawResponse() it will discard the intermediate FirebaseCall object, I don't think that's necessary a problem but ideally I'd like user to be able to worry only about:
- the `Firebase` type and its methods
- extracting standard result type from the API method call (int, float, string, json).